document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -114,7 +114,7 @@ private static String prepareParameterizedStringForList(
 
   public static String parseNonCompositeFilter(
       String fieldName, String op, Object value, Builder paramsBuilder) {
-    String fullFieldName = prepareCast(prepareFieldDataAccessorExpr(fieldName), value);
+    String fullFieldName = prepareCast(prepareFieldAccessorExpr(fieldName).toString(), value);
```

Comment:
In SQL, both having clause and group by clause should match the field accessor pattern. This has been changed to support normal field expression other than functional/aggregation in having cluase.

See the below query. Chec how `price` is accessed using `->` instead of `->>`.

```
SELECT COUNT(DISTINCT CAST (document->>'quantity' AS NUMERIC) ) AS qty_count, 
document->'item' AS item, 
document->'price' AS price 
FROM testCollection 
GROUP BY document->'item',document->'price' 
HAVING (COUNT(DISTINCT CAST (document->>'quantity' AS NUMERIC) ) <= 10) AND (
CAST (document->'price' AS NUMERIC) > 5)
```

Reply:
Probably, adding this explanation in a comment (maybe here or at the top of the class, or in tests) would help.